store/sqlstore/post_store.go

```diff
@@ -1206,10 +1206,8 @@ func (s *SqlPostStore) search(teamId string, userId string, params *model.Search
 		}
 	}
 
-	// these chars have special meaning and can be treated as spaces
-	for _, c := range specialSearchChar {
```

Comment:
I'm not really sure why do you need to replace special characters with spaces. So, I'm basically removing this and just, when there's a special character in the terms, I wrap the term with quotes. Thoughts?

Reply:
I believe that removing this will also fix #14437

Reply:
The special characters part is a way to tokenize manually removing reserved characters both for postgres and mysql.